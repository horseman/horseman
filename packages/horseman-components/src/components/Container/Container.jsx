import PropTypes from "prop-types";
import styled from "styled-components";

const defaultSizes = {
  small: 768,
  medium: 992,
  large: 1200,
  xl: 1400,
  fauxWidth: 2500,
};

/**
 * Retreive the pixel width as a number based on the theme and size
 */
const getContainerWidth = (theme, size) => {
  const sizes = {
    ...defaultSizes,
    ...theme.containers,
  };

  return sizes[size];
};

const Container = styled.div`
  margin: 0 auto;
  padding: 0 1em;

  max-width: ${({ theme, baseFontSize, size }) =>
    `${getContainerWidth(theme, size) / baseFontSize}em`};
`;

Container.defaultProps = {
  baseFontSize: 16,
  size: "large",
};

Container.propTypes = {
  baseFontSize: PropTypes.number,
  size: PropTypes.string,
};

/**
 * @component
 */
export default Container;
