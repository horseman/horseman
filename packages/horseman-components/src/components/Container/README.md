The Container serves as a wrapper for content with a max-width applied. The
large container is applied by default. Override the built-in container widths
within your active theme.

### Large Width Example
```
<Container>
  Large Width Example
</Container>

<Container size="xl">
  XL example
</Container>
```
