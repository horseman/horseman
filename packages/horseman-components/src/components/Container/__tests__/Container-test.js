import React from "react";
import { shallow } from "enzyme";

import Container from "../";

describe("Container", () => {
  test("renders defaults correctly", () => {
    const wrapper = shallow(<Container theme="undefined">Foo</Container>);
    expect(wrapper).toMatchSnapshot();
  });

  test("Can render with a custom baseFontSize", () => {
    const wrapper = shallow(<Container baseFontSize={14}>Foo</Container>);
    expect(wrapper).toMatchSnapshot();
  });

  test("Can render with custom size", () => {
    const wrapper = shallow(<Container size="medium">Foo</Container>);
    expect(wrapper).toMatchSnapshot();
  });

  test("Can render with theme size", () => {
    const wrapper = shallow(
      <Container
        theme={{
          containers: {
            custom: 500,
          },
        }}
        size="custom"
      >
        Foo
      </Container>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
