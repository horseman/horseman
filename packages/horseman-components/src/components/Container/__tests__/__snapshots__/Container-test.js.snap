// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Can render with a custom baseFontSize 1`] = `
.c0 {
  margin: 0 auto;
  padding: 0 1em;
  max-width: 85.71428571428571em;
}

<div
  className="c0"
  size="large"
>
  Foo
</div>
`;

exports[`Container Can render with custom size 1`] = `
.c0 {
  margin: 0 auto;
  padding: 0 1em;
  max-width: 62em;
}

<div
  className="c0"
  size="medium"
>
  Foo
</div>
`;

exports[`Container Can render with theme size 1`] = `
.c0 {
  margin: 0 auto;
  padding: 0 1em;
  max-width: 31.25em;
}

<div
  className="c0"
  size="custom"
>
  Foo
</div>
`;

exports[`Container renders defaults correctly 1`] = `
.c0 {
  margin: 0 auto;
  padding: 0 1em;
  max-width: 75em;
}

<div
  className="c0"
  size="large"
>
  Foo
</div>
`;
